compiler/noirc_evaluator/src/ssa/opt/flatten_cfg.rs

```diff
@@ -1,3 +1,27 @@
+//! This file contains the SSA flattening pass - a required pass for ACIR to remove any remaining
+//! control-flow in the singular program-function, resulting in a single block containing the
+//! program logic.
+//!
+//! ACIR/Brillig differences within this pass:
+//!   - This pass is strictly ACIR-only and never mutates brillig functions.
+//!
+//! Conditions:
+//!   - Precondition: There is at most one constrained function (inlining has been performed).
```

Comment:
This isn't strictly true as we have ACIR calls for use with folding, it's then either 0 ACIR functions if main entrypoint is a brillig function or >=1 if main entrypoint is ACIR function.